port/unix/omrsysinfo.c

```diff
@@ -2004,6 +2006,79 @@ omrsysinfo_get_OS_version(struct OMRPortLibrary *portLibrary)
 					rc = sysctlbyname(KERN_OSPRODUCTVERSION, buffer, &resultSize, NULL, 0);
 				}
 				buffer[resultSize] = '\0';
+
+				/* 10.16 is the version returned when in a backwards compatibility context in OSX versions 11+
+				* so we check the plink symlink as that bypasses the compatibility context and returns the
+				* real version info.
+				*/
+				if (0 == strncmp(buffer, "10.16", 5)) {
+					portLibrary->mem_free_memory(portLibrary, buffer);
+					buffer = NULL;
+					len = 0;
+					portLibrary->mem_free_memory(portLibrary, allocatedFileBuffer);
+					allocatedFileBuffer = NULL;
+
+					intptr_t plistFile = portLibrary->file_open(portLibrary, COMPAT_PLIST_LINK, EsOpenRead, 0444);
+					if (plistFile >= 0) {
+						char myBuffer[1024];  /* the file size is ~500 bytes, 525 on a 11.3 */
```

Comment:
Might as well remove this to save stack and use the previous definition of myBuffer, I think it's still in scope.

Reply:
Doesn't look like it. The previous definition of `myBuffer` is bound to the previous read of the plist file if sysctl failed.

Reply:
Ya ok, I see it now.